### ISSUE TITLE
core: Trouble running in Docker containers (or binding to 0.0.0.0)

### ISSUE THREAD

**Issue by mholt** [mholt/caddy#13]
It appears that specifying a 0.0.0.0 (wildcard) host is a special case that the virtual host feature doesn't handle yet. For example, 0.0.0.0:8080 should respond to anything that talked to port 8080.

Caddy responds when specifying just a port (":8080") but it's an error response.

**Comment by joeshaw** [comment]
You might want to go the nginx route of having a `default_server` that acts as a catch-all fallback when the `Host` header doesn't match any of the specified virtual hosts.

**Comment by joeshaw** [comment]
That'd let you separate the concerns of what IP address/interface the server binds to (ie, 0.0.0.0) versus the virtual hosts that it serves.

**Comment by abiosoft** [comment]
I have experienced this on my local pc without docker.

**Comment by darron** [comment]
Thanks for fixing this!

I created a Docker image for Caddy - https://github.com/darron/caddy-docker - but in order to do that and make a really small image I needed to statically link the final binary. The final Docker image was 23MB - vs using Ubuntu as a base image which was 148MB.

If possible, using your binaries would be better - people trust the author more than some random person compiling it - but you'd need to build them a little differently and statically link them.

Would that be possible for you to look at doing?

Thanks!
